src/hooks.ts

```diff
@@ -544,12 +591,13 @@ export const useTrackEvent: UseTrackEvent = () => {
     //    and we need to wait for the promise and update decision.
     // 2. When client is using datafile only but client is not ready yet which means user
     //    was provided as a promise and we need to subscribe and wait for user to become available.
-    if (optimizely.getIsUsingSdkKey() || !isClientReady) {
+    if (optimizely && (optimizely.getIsUsingSdkKey() || !isClientReady)) {
       subscribeToInitialization(optimizely, timeout, (initState) => {
         setState(initState);
       });
     }
-  }, []);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [optimizely, timeout]);
```

Comment:
I'm curious: why not include `isClientReady` here and above in the other hooks?

Reply:
Thats a great question. If client gets ready, `isClientReady` becomes `true` and triggers another execution of the `subscribeToInitialization` which is totally unnecessary in my opinion, because if client gets ready, there is no need to subscribe to anything after that. 

On the contrary, I could include the dep in the dependency array and that will cause one extra render, which is not terrible for sure. 

This is one of those rare cases where I feel like I can ignore the principles of "React".  But do let me know if you feel strongly against this. I don't mind changing it.

Reply:
Thanks for the teaching.